bootique-jetty/src/main/java/io/bootique/jetty/cors/BootiqueCorsFactory.java

```diff
@@ -0,0 +1,148 @@
+
+    private String allowedOrigins;
+    private String allowedTimingOrigins;
+    private String allowedMethods;
+    private String allowedHeaders;
+    private int preflightMaxAge;
+    private boolean allowCredentials;
+    private String exposedHeaders;
+    private boolean chainPreflight;
+    private int order;
+
+    public BootiqueCorsFactory(){
+
+        this.urlPatterns = new HashSet<>();
+        this.allowedOrigins = "*";
+        this.allowedTimingOrigins = "*";
+        this.allowedMethods = "X-Requested-With,Content-Type,Accept,Origin";
+        this.allowedHeaders = "GET,POST,HEAD";
+        this.preflightMaxAge = 1800;
+        this.allowCredentials = true;
+        this.exposedHeaders = "";
+        this.chainPreflight = true;
+        this.order = 1;
+    }
+
+    public String getAllowedOrigins() {
+        return allowedOrigins;
+    }
+
+    @BQConfigProperty
+    public void setAllowedOrigins(String allowedOrigins) {
+        this.allowedOrigins = allowedOrigins;
+    }
+
+    public String getAllowedTimingOrigins() {
+        return allowedTimingOrigins;
+    }
+
+    @BQConfigProperty
+    public void setAllowedTimingOrigins(String allowedTimingOrigins) {
+        this.allowedTimingOrigins = allowedTimingOrigins;
+    }
+
+    public String getAllowedMethods() {
+        return allowedMethods;
+    }
+
+    @BQConfigProperty
+    public void setAllowedMethods(String allowedMethods) {
+        this.allowedMethods = allowedMethods;
+    }
+
+    public String getAllowedHeaders() {
+        return allowedHeaders;
+    }
+
+    @BQConfigProperty
+    public void setAllowedHeaders(String allowedHeaders) {
+        this.allowedHeaders = allowedHeaders;
+    }
+
+    public int getPreflightMaxAge() {
+        return preflightMaxAge;
+    }
+
+    @BQConfigProperty
+    public void setPreflightMaxAge(int preflightMaxAge) {
+        this.preflightMaxAge = preflightMaxAge;
+    }
+
+    public boolean isAllowCredentials() {
+        return allowCredentials;
+    }
+
+    @BQConfigProperty
+    public void setAllowCredentials(boolean allowCredentials) {
+        this.allowCredentials = allowCredentials;
+    }
+
+    public String getExposedHeaders() {
+        return exposedHeaders;
+    }
+
+    @BQConfigProperty
+    public void setExposedHeaders(String exposedHeaders) {
+        this.exposedHeaders = exposedHeaders;
+    }
+
+    public boolean isChainPreflight() {
+        return chainPreflight;
+    }
+
+    @BQConfigProperty
+    public void setChainPreflight(boolean chainPreflight) {
+        this.chainPreflight = chainPreflight;
+    }
+
+    public Set<String> getUrlPatterns() {
+        return urlPatterns;
+    }
+
+    @BQConfigProperty
+    public void setUrlPatterns(Set<String> urlPatterns) {
+        this.urlPatterns = urlPatterns;
+    }
+
+
+    public int getOrder() {
+        return order;
+    }
+
+    @BQConfigProperty
+    public void setOrder(int order) {
+        this.order = order;
+    }
+
```

Comment:
This can be a private method if this class is turned into a true factory per my comment above.